enhancements/machine-api/machine-api-usage-telemetry.md

```diff
@@ -0,0 +1,338 @@
+platform is bringing to our users. It will show the operational burden we’ve
+decreased on teams that use OpenShift.
+
+**Remediation is Disabled or Short Circuited**
+
+Epics: Proactive Support
+
+This metric will give us insight into when a particular cluster has too many
+unhealthy nodes for the MHC to do its job. A user may be expecting the MHC is
+running as normal, cleaning up broken hosts, when in reality it has been silently disabled.
+Any sustained disabling of the MHC should potentially trigger a proactive
+support engagement. Similar to the _Total Remediation Events_, if any clusters
+experience this simultaneously, we can help provide any potential outage info
+to our users.
+
+**Number of Nodes Covered by the Machine Health Check**
+
+Epics: Engineering Insights, Proactive Support
+
+Users might think they are utilizing the MHC correctly only to find out that
+it has been misconfigured and not performing its role. If users have a large
+cluster with the MHC enabled, but not covering any nodes, this might be an
+opportunity to reach out and see if we can help with their configuration.
+Having this data will give us insight to drive the user experience of the MHC.
+
+**Failed Node Cleanups**
+
+Epics: Proactive Support
+
+This metric will help us identify users that are having problems with their
+infrastructure that might go unnoticed. We should trigger an alert if the
+autoscaler cleans up more than X hosts. This alert would be useful for users
+that might not realize they’re having these infrastructure problems.
+This is another case where signals across multiple clusters might indicate
+that a cloud or network area is having problems.
+
+For example, Our internal CI infra team had multiple MachineSets configured
+with the autoscaler.  One MachineSet would scale up fine, The other wouldn’t.
+The machines would fail, alarms would fire, the autoscaler would dispose of
+the nodes that failed to come online silently.
+
+**Total Scale Events**
+
+Epics: Demonstration of Value
+
+Collect data on the total number of scale up and scale down events,
+possibly as two individual metrics. We could use this data to couple with
+instance pricing to demonstrate how much using the autoscaler saves our users.
+This can also serve as a good proxy for how many users are actually using the
+autoscaler in a meaningful way, not just who has turned it on.
+
+**Unable to Scale Events**
+
+Epics: Proactive Support
+
+Collect data for when an autoscaler determines it cannot scale. We probably
+want to report different categories based on the types of failures.
+Events related to min/max threshold on machinesets probably don’t need
+alerting, but the inability to scale for other reasons (eg, hitting max CPU threshold or
+memory, but haven’t hit max node count) is useful to collect. These events can
+show opportunities to inform users about resource limitations in their clusters.
+
+**Machines with Old Delete Timestamp**
```

Comment:
fwiw this slightly overlaps with https://github.com/openshift/machine-api-operator/blob/master/install/0000_90_machine-api-operator_04_alertrules.yaml#L23-L32